### ISSUE TITLE
why is it giving me a 'npx'  command not found...

### ISSUE THREAD

**Issue by gotexis** [Reggionick/s3-deploy#13]
```
/usr/local/bin/npx s3-deploy@1.4.0 ./** --bucket *** --region *** --cwd . --distId *** --etag --gzip xml,html,htm,js,css,ttf,otf,svg,txt --invalidate / --noCache --deleteRemoved true
Error: There was an error when attempting to execute the process '/usr/local/bin/npx'. This may indicate the process failed to start. Error: spawn /usr/local/bin/npx ENOENT
```

**Comment by Reggionick** [comment]
On which container are you running the job? Seems that NodeJS is missing ore there is a wrong (old) version.

**Comment by samehhady** [comment]
facing the same issue, running on ubuntu-latest

`name: Deploy
on:
  push:
    branches: [ main ]
  pull_request:
    branches: [ main ]
defaults:
  run:
    working-directory: frontend
jobs:
  run:
    runs-on: ubuntu-latest
    env:
      AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
      AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
    steps:
        - uses: actions/checkout@v2

        - name: Install dependencies
          run: yarn

        - name: Build
          run: yarn build

        - name: Deploy
          uses: reggionick/s3-deploy@v3
          with:
            folder: build
            bucket: ${{ secrets.S3_BUCKET }}
            bucket-region: ${{ secrets.S3_BUCKET_REGION }}
            dist-id: ${{ secrets.CLOUDFRONT_DISTRIBUTION_ID }}
            invalidation: /
            delete-removed: true
            no-cache: true
            private: true
`

**Comment by samehhady** [comment]
/usr/local/bin/npx s3-deploy@1.4.0 ./** --bucket *** --region *** --cwd ./ --distId *** --etag --gzip xml,html,htm,js,css,ttf,otf,svg,txt --invalidate / --deleteRemoved --noCache --private
Error: There was an error when attempting to execute the process '/usr/local/bin/npx'. This may indicate the process failed to start. Error: spawn /usr/local/bin/npx ENOENT

**Comment by samehhady** [comment]
Found the issue, it is related to the "working-directory" when i set a default directory it fail, had to specify it in each step instead

**Comment by Reggionick** [comment]
Thanks @samehhady for the analysis. I close the issue since is not related to this action.

**Comment by CharlieDigital** [comment]
@samehhady Can you share some more details?  I'm facing the same issue:

```
Run reggionick/s3-deploy@v3.1.0
  with:
    folder: build
    bucket: ***
    bucket-region: ***
    delete-removed: true
    no-cache: true
    private: true
  env:
    AWS_ACCESS_KEY_ID: ***
    AWS_SECRET_ACCESS_KEY: ***
/opt/hostedtoolcache/node/12.20.0/x64/bin/npx s3-deploy@1.4.0 ./** --bucket *** --region *** --cwd ./ --etag --gzip xml,html,htm,js,css,ttf,otf,svg,txt --deleteRemoved --noCache --private
Error: There was an error when attempting to execute the process '/opt/hostedtoolcache/node/12.20.0/x64/bin/npx'. This may indicate the process failed to start. Error: spawn /opt/hostedtoolcache/node/12.20.0/x64/bin/npx ENOENT
```

**Comment by loginov-rocks** [comment]
Hi!

I guess it makes sense to respect the default parameters in the action :)

**Comment by vinicius5581** [comment]
> @samehhady Can you share some more details? I'm facing the same issue:
> 
> ```
> Run reggionick/s3-deploy@v3.1.0
>   with:
>     folder: build
>     bucket: ***
>     bucket-region: ***
>     delete-removed: true
>     no-cache: true
>     private: true
>   env:
>     AWS_ACCESS_KEY_ID: ***
>     AWS_SECRET_ACCESS_KEY: ***
> /opt/hostedtoolcache/node/12.20.0/x64/bin/npx s3-deploy@1.4.0 ./** --bucket *** --region *** --cwd ./ --etag --gzip xml,html,htm,js,css,ttf,otf,svg,txt --deleteRemoved --noCache --private
> Error: There was an error when attempting to execute the process '/opt/hostedtoolcache/node/12.20.0/x64/bin/npx'. This may indicate the process failed to start. Error: spawn /opt/hostedtoolcache/node/12.20.0/x64/bin/npx ENOENT
> ```

Did you ever figured this out? I'm getting the same error. 

```
Run reggionick/s3-deploy@v3
  with:
    folder: frontend/build
    bucket: ***
    bucket-region: ***
    dist-id: ***
    invalidation: /
    delete-removed: true
    no-cache: true
    private: true
  env:
    AWS_REGION: ***
    AWS_ACCESS_KEY_ID: ***
    AWS_SECRET_ACCESS_KEY: ***
  
/opt/hostedtoolcache/node/16.16.0/x64/bin/npx s[3](https://github.com/amzware/amzware-seller/runs/7711017599?check_suite_focus=true#step:5:3)-deploy@1.[4](https://github.com/amzware/amzware-seller/runs/7711017599?check_suite_focus=true#step:5:4).0 ./** --bucket *** --region *** --cwd ./ --distId *** --etag --gzip xml,html,htm,js,css,ttf,otf,svg,txt --invalidate / --deleteRemoved --noCache --private
Error: There was an error when attempting to execute the process '/opt/hostedtoolcache/node/1[6](https://github.com/PROJECTPATH/runs/7711017599?check_suite_focus=true#step:5:6).[16](https://github.com/PROJECTPATH/runs/7711017599?check_suite_focus=true#step:5:17).0/x64/bin/npx'. This may indicate the process failed to start. Error: spawn /opt/hostedtoolcache/node/16.16.0/x64/bin/npx ENOENT
```

**Comment by teerzo** [comment]
I'm getting the same error, im defining v3.2.0 but its reverting back to 1.40 because of a bug in the current npm version included for github actions 

it reverts back to the last published npm version (1.4.0 https://www.npmjs.com/package/s3-deploy)

Issue in npm has since been fixed, but github actions by default still uses the broken version 8.15.0
https://github.com/npm/cli/issues/5132#issuecomment-1224216976

Solution is to set the npm version yourself  
https://github.com/actions/runner-images/issues/6136#issuecomment-1225422706

Hope this helps
